pkg/operation/hybridbotanist/controlplane.go

```diff
@@ -521,15 +520,26 @@ func (b *HybridBotanist) DeployETCD(ctx context.Context) error {
 		}
 
 		if b.Shoot.HibernationEnabled {
-			statefulset := &appsv1.StatefulSet{}
```

Comment:
Why is this code removed? I think we should still have it, and move the whole `if lastSnapshotRevision > 0 && ...` part into the `if statefulset.Spec.Replicas == nil` section.
WDYT?

Reply:
When we know replicas could be 1 or 0. I didn't get why did we had unnecessary `Get` call there.  So, removed it.

Reply:
It's because of hibernation - when a shoot is hibernated the etcd is only scaled down at the very end of the flow.
Please revert the behaviour.

Reply:
Oh OK. I was unaware of it. For me change worked. So, didn't noticed it. On it.